rpc/chain/client.go

```diff
@@ -977,15 +976,19 @@ func (c *ClientWithFallback) SetWalletNotifier(notifier func(chainId uint64, mes
 
 func (c *ClientWithFallback) toggleConnectionState(err error) {
 	connected := true
+	skipNotification := false
 	if err != nil {
-		if !isNotFoundError(err) && !isVMError(err) && !errors.Is(err, ErrRequestsOverLimit) && !errors.Is(err, context.Canceled) {
+		if network_utils.IsConnectionError(err) {
+			connected = false
+			skipNotification = true
```

Comment:
I'm still not happy about not sending the notifications in this case...
Example, some providers could be inaccessible due to firewall rules, while others could be working just fine. The client should be able to say "hey, you cannot access these providers so you cannot do this and this operations, but you can do these other ones for now".

What I would do, if you feel it's reasonable, is add some enum field to the notification with more detail about the cause for the "provider down" event. One of the values could be "Network Down" which I feel would have the same value as the `skipNotification` flag here. Then, if the Mobile client wants to not raise banners when that's the reason, you guys can do that.

Reply:
As I mentioned here https://github.com/status-im/status-go/pull/5820#discussion_r1763086556

What do you think about making it a separate task? 
Because it will require some clarification from the UI team.

Reply:
Sorry, perhaps I'm being stubborn and I don't want to block you unnecessarily so I'd like to hear some other folks' opinions.

The way I see it right now, this could happen after these changes:
- Chain providers are down for whatever reason (firewall, providers are really down, etc), so we cannot refresh balances. We inhibit the notification.
- The completely independent "Internet connection down" state (I think it's tied to the number of Waku peers being 0 or something) does not get triggered
- The client shows no banner about providers being down, internet connection being down or balances being out of date

Reply:
Added some findings while I was working on this task https://github.com/status-im/status-go/issues/5854